reporter/src/main/kotlin/gitlab/LicenseModelMapper.kt

```diff
@@ -0,0 +1,122 @@
+    fun map(ortResult: OrtResult, skipExcluded: Boolean): GitlabLicenseModel {
+        val packagesWithDefinitionFilePaths = ortResult.getTargetPackagesWithDefinitionFiles(skipExcluded)
+
+        return GitlabLicenseModel(
+            licenses = packagesWithDefinitionFilePaths.keys.toLicenses(),
+            dependencies = packagesWithDefinitionFilePaths.toDependencies()
+        )
+    }
+}
+
+private fun Collection<Package>.toLicenses(): List<License> {
+    val decomposedLicenses = flatMapTo(mutableSetOf()) { pkg ->
+        pkg.declaredLicensesProcessed.spdxExpression?.decompose().orEmpty()
+    }
+
+    return decomposedLicenses.map { singleLicenseExpression ->
+        License(
+            id = singleLicenseExpression.simpleLicense(),
+            name = singleLicenseExpression.toLicenseName(),
+            url = "" // TODO
+        )
+    }.sortedBy { it.id }
+}
+
+private fun Map<Package, List<String>>.toDependencies(): List<Dependency> =
+    map { (pkg, definitionFilePaths) ->
+        pkg.toDependency(definitionFilePaths)
+    }.sortedBy { "${it.packageManger}${it.name}${it.version}" }
+
+private fun OrtResult.getTargetPackagesWithDefinitionFiles(skipExcluded: Boolean): Map<Package, List<String>> {
+    val result = mutableMapOf<Identifier, MutableList<String>>()
+
+    getProjects(omitExcluded = skipExcluded).forEach { project ->
+        val definitionFilePath = project.definitionFilePath
+
+        // Omit non-referenced and maybe also excluded packages:
+        project.collectDependencies { !skipExcluded || !isExcluded(it.id) }.forEach { id ->
+            result.getOrPut(id) { mutableListOf() } += definitionFilePath
+        }
+    }
+
+    return result.mapKeys { getPackage(it.key)!!.pkg }
+}
+
+private fun Package.toDependency(definitionFilePaths: Collection<String>): Dependency =
+    Dependency(
+        name = id.name,
+        version = id.version,
+        licenses = declaredLicensesProcessed.spdxExpression?.decompose().orEmpty().map { it.toString() },
+        path = definitionFilePaths.sorted().joinToString(","),
+        packageManger = id.toPackageManagerName()
+    )
+
+private fun SpdxSingleLicenseExpression.toLicenseName(): String {
+    // TODO: Get the full name also for non-SPDX / ScanCode licenses.
+    val spdxLicenseId = when (this) {
+        is SpdxLicenseWithExceptionExpression -> license.simpleLicense()
+        is SpdxLicenseIdExpression -> this.simpleLicense()
+        else -> ""
+    }
+
+    return SpdxLicense.forId(spdxLicenseId)?.fullName ?: ""
+}
+
+/**
+ * Map the [Identifier.type] to the Gitlab license model package names. The mapping is based on the examples found
+ * under https://gitlab.com/gitlab-org/security-products/license-management/-/tree/master/spec/fixtures/expected.
+ */
+private fun Identifier.toPackageManagerName(): String =
+    when (type) {
+        "Bower" -> "bower"
+        "Bundler" -> "bundler"
+        "Composer" -> "composer"
+        "Conan" -> "Conan"
```

Comment:
https://gitlab.com/gitlab-org/security-products/license-management/-/blob/ab794b1050830a01f6fede765e9eb40cdbfcdc38/spec/fixtures/expected/c/conan/v2.1.json#L14 looks like this should map to lower-case "conan" (please also double-check the "Go" mappings below).

Reply:
Good catch with the "Conan" entry. 
What do you mean by double checking the "Go" mappings? I checked them already. Do you have any indication these are wrong.

Reply:
Done.